src/Symfony/Bundle/FrameworkBundle/Command/SecretsListCommand.php

```diff
@@ -45,12 +46,15 @@ public function __construct(AbstractVault $vault, AbstractVault $localVault = nu
     protected function configure(): void
     {
         $this
+            ->addArgument('name', InputArgument::OPTIONAL, 'The name of the secret to return')
```

Comment:
You can add value completion with the list of known list of names.

Reply:
I don't quite get what you mean. Do you mean:

```php
protected function configure(): void
{
    $this
        ->addArgument('name', InputArgument::OPTIONAL, 'The name of the secret to return', null, \array_keys($this->vault->list()))

Reply:
related to this for example https://github.com/symfony/symfony/blob/4328f5b16852b8daefc957d370c08ee0bfbf3dbc/src/Symfony/Bundle/FrameworkBundle/Command/SecretsRemoveCommand.php#L84

https://symfony.com/blog/new-in-symfony-5-4-console-autocompletion

Reply:
Isn't that in the end identical to what I suggested? By default the `complete` method uses the last argument passed to addArgument/addOption and adds them as suggested values. We don't need any special logic here for what secret names can be suggested or not.

Reply:
Yes @danielburger1337 your proposition is better as it's the new recommended way of doing it.

```php
    $this
        ->addArgument('name', InputArgument::OPTIONAL, 'The name of the secret to return', null, fn() => \array_keys($this->vault->list()))
```

A closure should be used for lazy loading.